core/state_processor.go

```diff
@@ -128,7 +128,7 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, bc ChainCon
 	// Set the receipt logs and create the bloom filter.
 	receipt.Logs = statedb.GetLogs(tx.Hash())
```

Comment:
What about setting `Log.BlockHash` here instead of `AddLog`?

Reply:
You mean like this: 
```
diff --git a/core/state_processor.go b/core/state_processor.go
index 0cfac2975..835e6740a 100644
--- a/core/state_processor.go
+++ b/core/state_processor.go
@@ -126,9 +126,14 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, bc ChainCon
        }
 
        // Set the receipt logs and create the bloom filter.
-       receipt.Logs = statedb.GetLogs(tx.Hash())
+       blockhash := header.Hash()
+       logs := statedb.GetLogs(tx.Hash())
+       for _, l := range logs {
+               l.BlockHash = blockhash
+       }
+       receipt.Logs = logs
        receipt.Bloom = types.CreateBloom(types.Receipts{receipt})
-       receipt.BlockHash = header.Hash()
+       receipt.BlockHash = blockhash
        receipt.BlockNumber = header.Number
        receipt.TransactionIndex = uint(statedb.TxIndex())
        return receipt, err
        ```
        I'm not sure if we use the blockhash in the receipts ever, but I would be fine with adding this